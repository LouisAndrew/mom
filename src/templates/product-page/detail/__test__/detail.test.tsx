import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import Detail from '..';

describe('Detail on product page', () => {
    const mockProperty = {
        name: 'Rumah 1',
        address: 'Jalan rumah 1',
        propertyType: 'house',
        saleType: 'sell',
        description: 'Deskripsi bum bum bum',
        location: 'Penang',
        price: 200,
        area: 125,
        bathQty: 2,
        roomQty: 3,
    };
    const el: React.ReactElement = <Detail {...mockProperty} />;
    afterEach(cleanup);

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(el, div);
    });

    it('matches snapshot', () => {
        const tree = renderer.create(el).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
