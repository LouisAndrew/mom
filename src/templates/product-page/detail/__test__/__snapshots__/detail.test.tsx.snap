// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail on product page matches snapshot 1`] = `
<div
  className="sc-fzpmMD fnAHmJ"
  display="flex"
>
  <h1
    className="sc-AxjAm bKBrqz"
    fontFamily="heading"
    fontSize={
      Array [
        5,
        5,
        6,
        6,
      ]
    }
    fontWeight="extraBold"
  >
    Rumah 1
    <div
      className="sc-fzoXWK izQThk"
    >
      <span
        style={
          Object {
            "border": "2px solid #007bff",
            "color": "#007bff",
          }
        }
      >
        Jual
      </span>
      <span
        style={
          Object {
            "border": "2px solid #28a745",
            "color": "#28a745",
          }
        }
      >
        Rumah
      </span>
    </div>
  </h1>
  <h3
    className="sc-AxiKw hMdbcy"
    fontFamily="heading"
    fontSize={
      Array [
        3,
        3,
        4,
        4,
      ]
    }
    width={1}
  >
    Jalan rumah 1
    <span
      style={
        Object {
          "border": "2px solid #587B7F",
          "borderRadius": 4,
          "color": "#587B7F",
          "marginLeft": 32,
          "padding": "4px 8px",
        }
      }
    >
      Penang
    </span>
  </h3>
  <div
    className="sc-fzoXWK dDTsOr"
  >
    <p>
      Luas
    </p>
    <h3
      className="sc-AxiKw hvMtGq"
      fontFamily="heading"
      fontSize={
        Array [
          3,
          3,
          4,
          4,
        ]
      }
    >
      <svg
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M3 19a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2H5a2 2 0 0 0-2 2v14zm9-13h6v6h-2V8h-4V6zm-6 6h2v4h4v2H6v-6z\\" fill=\\"currentColor\\"/>",
          }
        }
        focusable={false}
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "transform": "rotate(360deg)",
            "verticalAlign": "-0.125em",
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      />
      125
    </h3>
    <p>
      Kamar tidur
    </p>
    <h3
      className="sc-AxiKw hvMtGq"
      fontFamily="heading"
      fontSize={
        Array [
          3,
          3,
          4,
          4,
        ]
      }
    >
      <svg
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M20 9.556V3h-2v2H6V3H4v6.557C2.81 10.25 2 11.526 2 13v4a1 1 0 0 0 1 1h1v4h2v-4h12v4h2v-4h1a1 1 0 0 0 1-1v-4c0-1.474-.811-2.75-2-3.444zM11 9H6V7h5v2zm7 0h-5V7h5v2z\\" fill=\\"currentColor\\"/>",
          }
        }
        focusable={false}
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "transform": "rotate(360deg)",
            "verticalAlign": "-0.125em",
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      />
      3
    </h3>
    <p>
      Kamar mandi
    </p>
    <h3
      className="sc-AxiKw hvMtGq"
      fontFamily="heading"
      fontSize={
        Array [
          3,
          3,
          4,
          4,
        ]
      }
    >
      <svg
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M21 10H7V7.113c0-.997.678-1.923 1.661-2.085A2.003 2.003 0 0 1 11 7h2a4.003 4.003 0 0 0-4.398-3.98C6.523 3.222 5 5.089 5 7.178V10H3a1 1 0 0 0-1 1v2c0 2.606 1.674 4.823 4 5.65V22h2v-3h8v3h2v-3.35c2.326-.827 4-3.044 4-5.65v-2a1 1 0 0 0-1-1z\\" fill=\\"currentColor\\"/>",
          }
        }
        focusable={false}
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "transform": "rotate(360deg)",
            "verticalAlign": "-0.125em",
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      />
      2
    </h3>
  </div>
  <h2
    className="sc-AxirZ iLnAXc"
    fontFamily="heading"
    fontSize={
      Array [
        4,
        4,
        5,
        5,
      ]
    }
  >
    200 Jt
  </h2>
  <button
    className="sc-AxheI hMzLqx"
    fontFamily="body"
    fontSize={
      Array [
        1,
        1,
        2,
        2,
      ]
    }
    fontWeight="bold"
    onClick={[Function]}
    width={1}
  >
    <svg
      dangerouslySetInnerHTML={
        Object {
          "__html": "<path d=\\"M16.6 14c-.2-.1-1.5-.7-1.7-.8c-.2-.1-.4-.1-.6.1c-.2.2-.6.8-.8 1c-.1.2-.3.2-.5.1c-.7-.3-1.4-.7-2-1.2c-.5-.5-1-1.1-1.4-1.7c-.1-.2 0-.4.1-.5c.1-.1.2-.3.4-.4c.1-.1.2-.3.2-.4c.1-.1.1-.3 0-.4c-.1-.1-.6-1.3-.8-1.8c-.1-.7-.3-.7-.5-.7h-.5c-.2 0-.5.2-.6.3c-.6.6-.9 1.3-.9 2.1c.1.9.4 1.8 1 2.6c1.1 1.6 2.5 2.9 4.2 3.7c.5.2.9.4 1.4.5c.5.2 1 .2 1.6.1c.7-.1 1.3-.6 1.7-1.2c.2-.4.2-.8.1-1.2l-.4-.2m2.5-9.1C15.2 1 8.9 1 5 4.9c-3.2 3.2-3.8 8.1-1.6 12L2 22l5.3-1.4c1.5.8 3.1 1.2 4.7 1.2c5.5 0 9.9-4.4 9.9-9.9c.1-2.6-1-5.1-2.8-7m-2.7 14c-1.3.8-2.8 1.3-4.4 1.3c-1.5 0-2.9-.4-4.2-1.1l-.3-.2l-3.1.8l.8-3l-.2-.3c-2.4-4-1.2-9 2.7-11.5S16.6 3.7 19 7.5c2.4 3.9 1.3 9-2.6 11.4\\" fill=\\"currentColor\\"/>",
        }
      }
      focusable={false}
      height="1em"
      preserveAspectRatio="xMidYMid meet"
      style={
        Object {
          "transform": "rotate(360deg)",
          "verticalAlign": "-0.125em",
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    />
    Hubungi Saya
  </button>
</div>
`;
